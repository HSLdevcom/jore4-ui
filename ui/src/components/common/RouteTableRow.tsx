import { gql } from '@apollo/client';
import { RouteTableRowFragment } from '../../generated/graphql';
import {
  useAppDispatch,
  useAppSelector,
  useShowRoutesOnModal,
} from '../../hooks';
import {
  deselectRouteLabelAction,
  selectExport,
  selectRouteLabelAction,
} from '../../redux';
import { RouteLineTableRow } from './RouteLineTableRow';

const GQL_ROUTE_TABLE_ROW = gql`
  fragment route_table_row on route_route {
    ...route_information_for_map
    name_i18n
    direction
    priority
    on_line_id
  }
`;

interface Props {
  className?: string;
  route: RouteTableRowFragment;
  linkTo: string;
  isSelectable?: boolean;
}

/**
 * Reusable component RouteTableRow for list views. This component requires
 * the route information (RouteTableRowFragment) and linkTo parameter to
 * determine where we navigate after clicking this row.
 */
export const RouteTableRow = ({
  className = '',
  route,
  linkTo,
  isSelectable = false,
}: Props): JSX.Element => {
  const { showRouteOnMapByLabel } = useShowRoutesOnModal();
  const dispatch = useAppDispatch();
  const { selectedRouteLabels } = useAppSelector(selectExport);

  const onClickShowRouteOnMap = () => {
    showRouteOnMapByLabel(route);
  };

  const onSelectChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selected = event.target.checked;
    const selectAction = selected
      ? selectRouteLabelAction
      : deselectRouteLabelAction;

    dispatch(selectAction(route.label));
  };

  const isSelected = selectedRouteLabels.includes(route.label);

  return (
    <RouteLineTableRow
      rowItem={route}
      linkTo={linkTo}
      onLocatorButtonClick={
        route.route_shape /* some routes imported from jore3 are missing the geometry */
          ? onClickShowRouteOnMap
          : undefined
      }
      locatorButtonTestId="RouteTableRow::showRoute"
      className={className}
      isSelected={isSelected}
      onSelectChanged={isSelectable ? onSelectChanged : undefined}
    />
  );
};
