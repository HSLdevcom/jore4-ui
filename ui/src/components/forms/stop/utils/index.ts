export * from './formatIsoDateString';
export * from './useFindStopAreas';
