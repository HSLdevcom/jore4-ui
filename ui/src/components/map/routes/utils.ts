export const NEW_ROUTE_LINE_ID = 'new_route_line';
export const NEW_ROUTE_ARROWS_ID = 'new_route_arrows';

const ROUTE_LAYER_ID_PREFIX = 'route_id_';
// Get route id between route layer id prefix and _line/_arrows postfix
const ROUTE_ID_FROM_LAYER_ID_REGEX = new RegExp(
  `(?<=${ROUTE_LAYER_ID_PREFIX})(.*)(?=_)`,
);

// Utilities to construct layer ids from route ids
export function mapRouteIdToLineLayerId(routeId: UUID) {
  return `${ROUTE_LAYER_ID_PREFIX}${routeId}_line`;
}

export function mapRouteIdToArrowLayerId(routeId: UUID) {
  return `${ROUTE_LAYER_ID_PREFIX}${routeId}_arrows`;
}

// Utilities to allow finding the original route's id based on the layer's id
export function isRouteGeometryLayer(layerId: string) {
  return layerId.startsWith(ROUTE_LAYER_ID_PREFIX);
}

export function mapLayerIdToRouteId(layerId: string) {
  const matches = layerId.match(ROUTE_ID_FROM_LAYER_ID_REGEX);

  return matches?.length ? (matches[0] as UUID) : undefined;
}
