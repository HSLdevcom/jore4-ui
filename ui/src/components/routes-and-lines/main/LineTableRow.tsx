import { gql } from '@apollo/client';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { LineTableRowFragment } from '../../../generated/graphql';
import { useAlertsAndHighLights, useShowRoutesOnModal } from '../../../hooks';
import { Column, Row } from '../../../layoutComponents';
import { Path, routeDetails } from '../../../router/routeDetails';
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../../../time';
import { LocatorButton } from '../../../uiComponents';

interface Props {
  className?: string;
  line: LineTableRowFragment;
}

const GQL_LINE_TABLE_ROW = gql`
  fragment line_table_row on route_line {
    ...line_all_fields
    line_routes {
      route_id
    }
  }
`;

export const LineTableRow = ({ className = '', line }: Props): JSX.Element => {
  const { t } = useTranslation();
  const { showRoutesOnModal } = useShowRoutesOnModal();

  const showLineRoutes = () => {
    const lineRouteIds = line.line_routes?.map((item) => item.route_id);

    showRoutesOnModal(
      lineRouteIds,
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      line.validity_start!,
      line.validity_end,
    );
  };

  const { getAlertLevel, getAlertStyle } = useAlertsAndHighLights();
  const alertStyle = getAlertStyle(getAlertLevel(line));

  return (
    <tbody>
      <tr className={`border border-light-grey ${className}`}>
        <td className={`${alertStyle.listItemBorder || ''} w-20 p-4 align-top`}>
          {alertStyle.icon && <i className={`${alertStyle.icon} text-3xl`} />}
        </td>
        <td className="py-4">
          <Link to={routeDetails[Path.lineDetails].getLink(line.line_id)}>
            <Row className="items-center">
              <Column className="w-1/2 font-bold">
                <p className="text-3xl">{line.label}</p>
                <p>{line.name_i18n.fi_FI}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(line.validity_start || MIN_DATE),
                    endDate: mapToShortDate(line.validity_end || MAX_DATE),
                  })}
                </p>
                <p>!Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija</p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="w-20 p-6 align-middle">
          <LocatorButton
            onClick={showLineRoutes}
            testId="LineTableRow::showLineRoutes"
          />
        </td>
      </tr>
    </tbody>
  );
};
