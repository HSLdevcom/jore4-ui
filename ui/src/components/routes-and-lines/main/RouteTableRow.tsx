import { gql } from '@apollo/client';
import { RouteTableRowFragment } from '../../../generated/graphql';
import {
  useAppDispatch,
  useAppSelector,
  useShowRoutesOnModal,
} from '../../../hooks';
import {
  deselectRouteLabelAction,
  selectExport,
  selectRouteLabelAction,
} from '../../../redux';
import { RouteLineTableRow } from './RouteLineTableRow';

const GQL_ROUTE_TABLE_ROW = gql`
  fragment route_table_row on route_route {
    ...route_information_for_map
    name_i18n
    direction
    priority
    on_line_id
  }
`;

interface Props {
  className?: string;
  route: RouteTableRowFragment;
  isSelectable?: boolean;
}

export const RouteTableRow = ({
  className = '',
  route,
  isSelectable = false,
}: Props): JSX.Element => {
  const { showRouteOnMapByLabel } = useShowRoutesOnModal();
  const dispatch = useAppDispatch();
  const { selectedRouteLabels } = useAppSelector(selectExport);

  const onClickShowRouteOnMap = () => {
    showRouteOnMapByLabel(route);
  };

  const onSelectChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selected = event.target.checked;
    const selectAction = selected
      ? selectRouteLabelAction
      : deselectRouteLabelAction;

    dispatch(selectAction(route.label));
  };

  const isSelected = selectedRouteLabels.includes(route.label);

  return (
    <RouteLineTableRow
      rowItem={route}
      lineId={route.on_line_id}
      onLocatorButtonClick={
        route.route_shape /* some routes imported from jore3 are missing the geometry */
          ? onClickShowRouteOnMap
          : undefined
      }
      locatorButtonTestId="RouteTableRow::showRoute"
      className={className}
      isSelected={isSelected}
      onSelectChanged={isSelectable ? onSelectChanged : undefined}
    />
  );
};
