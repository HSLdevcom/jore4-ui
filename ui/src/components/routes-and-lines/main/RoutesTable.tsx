import React from 'react';

interface Props {
  className?: string;
  testId?: string;
}

export const RoutesTable: React.FC<Props> = ({
  className,
  testId,
  children,
}) => {
  return (
    // setting a fake "height: 1px" so that "height: 100%" would work for the table cells
    <table className={`h-1 w-full ${className}`} data-testid={testId}>
      {children}
    </table>
  );
};
