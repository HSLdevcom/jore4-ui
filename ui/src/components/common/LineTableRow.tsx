import { gql } from '@apollo/client';
import { LineTableRowFragment } from '../../generated/graphql';
import {
  useAppDispatch,
  useAppSelector,
  useShowRoutesOnModal,
} from '../../hooks';
import {
  deselectRouteUniqueLabelsAction,
  selectExport,
  selectRouteUniqueLabelsAction,
} from '../../redux';
import { routeHasTimetables } from '../../utils/route';
import {
  RouteLineTableRow,
  RouteLineTableRowVariant,
} from './RouteLineTableRow';

interface Props {
  className?: string;
  line: LineTableRowFragment;
  isSelectable?: boolean;
  rowVariant: RouteLineTableRowVariant;
}

const GQL_LINE_TABLE_ROW = gql`
  fragment line_table_row on route_line {
    name_i18n
    short_name_i18n
    priority
    ...line_map_params
    line_routes {
      ...route_map_params
      unique_label
      route_journey_patterns {
        journey_pattern_id
        journey_pattern_refs {
          journey_pattern_ref_id
          vehicle_journeys {
            vehicle_journey_id
          }
        }
      }
    }
  }
`;

/**
 * Reusable component LineTableRow for list views. This component requires
 * the line information (LineTableRowFragment) and linkTo parameter to
 * determine where we navigate after clicking this row.
 */
export const LineTableRow = ({
  className = '',
  line,
  isSelectable = false,
  rowVariant,
}: Props): JSX.Element => {
  const { showRoutesOnMapByLineLabel } = useShowRoutesOnModal();
  const dispatch = useAppDispatch();
  const { selectedRouteUniqueLabels } = useAppSelector(selectExport);

  const showLineRoutes = () => {
    showRoutesOnMapByLineLabel(line);
  };

  const onSelectChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selected = event.target.checked;
    const selectAction = selected
      ? selectRouteUniqueLabelsAction
      : deselectRouteUniqueLabelsAction;

    dispatch(selectAction(line.line_routes.map((route) => route.label)));
  };

  const hasRoutes = line.line_routes?.length > 0;

  // Entire line is selected for export if it has routes and each of those is selected.
  // Selecting only subset of routes shouldn't be possible, as user can select routes
  // from line list or from route list, not from both simultaneously.
  const isSelected =
    hasRoutes &&
    line.line_routes.every((route) =>
      selectedRouteUniqueLabels.includes(route.label),
    );

  // Check if any of the line's route has timetables existing
  const hasTimetables = line.line_routes.some(routeHasTimetables);

  return (
    <RouteLineTableRow
      rowItem={line}
      rowVariant={rowVariant}
      hasTimetables={hasTimetables}
      onLocatorButtonClick={showLineRoutes}
      locatorButtonTestId="LineTableRow::showLineRoutes"
      className={className}
      lineId={line.line_id}
      onSelectChanged={isSelectable ? onSelectChanged : undefined}
      isSelected={isSelected}
      selectionDisabled={!hasRoutes}
    />
  );
};
