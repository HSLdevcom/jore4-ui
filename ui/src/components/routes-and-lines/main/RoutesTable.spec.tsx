import { buildLocalizedString, buildRoute } from '@hsl/jore4-test-db-manager';
import {
  ListChangingRoutesQuery,
  RouteDirectionEnum,
  RouteLine,
  RouteRoute,
  useListOwnLinesQuery,
} from '../../../generated/graphql';
import {
  GqlQueryResult,
  mapListOwnLinesResult,
  mapRouteResultToRoutes,
} from '../../../graphql';
import { render } from '../../../utils/test-utils';
import { LineTableRow } from './LineTableRow';
import { RoutesTable } from './RoutesTable';
import { RouteTableRow } from './RouteTableRow';

describe(`<${RoutesTable.name} />`, () => {
  const testId = 'routesTable1';
  const route1Id = '03d55414-e5cf-4cce-9faf-d86ccb7e5f98';
  const lineId = '101f800c-39ed-4d85-8ece-187cd9fe1c5e';

  // this response is copy-pasted from the actual graphql response
  const routesResponseMock: GqlQueryResult<ListChangingRoutesQuery> = {
    data: {
      route_route: [
        {
          __typename: 'route_route',
          ...buildRoute({ label: '1' }),
          route_id: route1Id,
          on_line_id: lineId,
          priority: 10,
          direction: RouteDirectionEnum.Outbound,
          route_line: {
            line_id: lineId,
            label: '65',
            name_i18n: buildLocalizedString(
              'Rautatientori - Kätilöopisto - Veräjälaakso',
            ),
            short_name_i18n: buildLocalizedString(
              'Rautatientori - Veräjälaakso',
            ),
            __typename: 'route_line',
          },
        },
      ],
    },
  };

  test('Renders the table with route data', async () => {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const routes = mapRouteResultToRoutes(routesResponseMock)!;
    const { asFragment } = render(
      <RoutesTable testId={testId}>
        {routes.map((item: RouteRoute) => (
          <RouteTableRow
            key={item.route_id}
            route={item}
            isSelectable={false}
          />
        ))}
      </RoutesTable>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // this response is copy-pasted from the actual graphql response
  const linesResponseMock = {
    data: {
      route_line: [
        {
          line_id: '101f800c-39ed-4d85-8ece-187cd9fe1c5e',
          label: '65',
          name_i18n: buildLocalizedString('Rautatientori - Veräjälaakso'),
          short_name_i18n: buildLocalizedString('Rautatientori - Veräjälaakso'),
          __typename: 'route_line',
        },
        {
          line_id: '9058c328-efdd-412c-9b2b-37b0f6a2c6fb',
          label: '71',
          name_i18n: buildLocalizedString('Rautatientori - Malmi as.'),
          short_name_i18n: buildLocalizedString('Rautatientori - Malmi as.'),
          __typename: 'route_line',
        },
        {
          line_id: 'bbd1cb29-74c3-4fa1-ac86-918d7fa96fe2',
          label: '785K',
          name_i18n: buildLocalizedString('Rautatientori - Nikkilä'),
          short_name_i18n: buildLocalizedString('Rautatientori - Nikkilä'),
          __typename: 'route_line',
        },
        {
          line_id: 'db748c5c-42e3-429f-baa0-e0db227dc2c8',
          label: '1234',
          name_i18n: buildLocalizedString('Erottaja - Arkkadiankatu'),
          short_name_i18n: buildLocalizedString('Erottaja - Arkkadiankatu'),
          __typename: 'route_line',
        },
      ],
    },
  } as ReturnType<typeof useListOwnLinesQuery>;

  test('Renders the table with line data', async () => {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const lines = mapListOwnLinesResult(linesResponseMock)!;
    const { asFragment } = render(
      <RoutesTable testId={testId}>
        {lines.map((item: RouteLine) => (
          <LineTableRow key={item.line_id} line={item} isSelectable={false} />
        ))}
      </RoutesTable>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
