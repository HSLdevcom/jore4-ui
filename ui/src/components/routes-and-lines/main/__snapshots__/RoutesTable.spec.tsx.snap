// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoutesTable /> Renders the table with line data 1`] = `
<DocumentFragment>
  <table
    class="h-1 w-full "
    data-testid="routesTable1"
  >
    <tbody>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class=""
            href="/lines/101f800c-39ed-4d85-8ece-187cd9fe1c5e"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    65
                  </h2>
                  <i
                    class="icon-bus-alt text-2xl text-tweaked-brand"
                  />
                </div>
                <p>
                  Rautatientori - Veräjälaakso
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
            data-testid="LocatorButton::button"
            disabled=""
            type="button"
          >
            <i
              class="icon-calendar inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand  "
              data-testid="LineTableRow::showLineRoutes"
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class=""
            href="/lines/9058c328-efdd-412c-9b2b-37b0f6a2c6fb"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    71
                  </h2>
                  <i
                    class="icon-bus-alt text-2xl text-tweaked-brand"
                  />
                </div>
                <p>
                  Rautatientori - Malmi as.
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
            data-testid="LocatorButton::button"
            disabled=""
            type="button"
          >
            <i
              class="icon-calendar inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand  "
              data-testid="LineTableRow::showLineRoutes"
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class=""
            href="/lines/bbd1cb29-74c3-4fa1-ac86-918d7fa96fe2"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    785K
                  </h2>
                  <i
                    class="icon-bus-alt text-2xl text-tweaked-brand"
                  />
                </div>
                <p>
                  Rautatientori - Nikkilä
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
            data-testid="LocatorButton::button"
            disabled=""
            type="button"
          >
            <i
              class="icon-calendar inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand  "
              data-testid="LineTableRow::showLineRoutes"
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class=""
            href="/lines/db748c5c-42e3-429f-baa0-e0db227dc2c8"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    1234
                  </h2>
                  <i
                    class="icon-bus-alt text-2xl text-tweaked-brand"
                  />
                </div>
                <p>
                  Erottaja - Arkkadiankatu
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
            data-testid="LocatorButton::button"
            disabled=""
            type="button"
          >
            <i
              class="icon-calendar inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand  "
              data-testid="LineTableRow::showLineRoutes"
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<RoutesTable /> Renders the table with route data (routes and lines) 1`] = `
<DocumentFragment>
  <table
    class="h-1 w-full "
    data-testid="routesTable1"
  >
    <tbody>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class=""
            href="/lines/101f800c-39ed-4d85-8ece-187cd9fe1c5e?routeLabels=1"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    <b>
                      1
                    </b>
                  </h2>
                  <i
                    class="icon-bus-alt text-2xl text-tweaked-brand"
                  />
                </div>
                <p>
                  route 1
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
            data-testid="LocatorButton::button"
            disabled=""
            type="button"
          >
            <i
              class="icon-calendar inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
              data-testid="RouteTableRow::showRoute"
              disabled=""
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<RoutesTable /> Renders the table with route data (timetables) 1`] = `
<DocumentFragment>
  <table
    class="h-1 w-full "
    data-testid="routesTable1"
  >
    <tbody>
      <tr
        class=""
      >
        <td
          class="border-l-10 border-hsl-dark-green border-y border-y-light-grey p-4"
        >
          <i
            class="icon-_-placeholder my-auto flex text-3xl"
          />
        </td>
        <td
          class="w-full py-4 border-y border-y-light-grey"
        >
          <a
            class="pointer-events-none text-zinc-400"
            href="/timetables/lines/101f800c-39ed-4d85-8ece-187cd9fe1c5e?routeLabels=1"
          >
            <div
              class="flex flex-row items-center"
            >
              <div
                class="flex flex-col w-1/2 font-bold"
              >
                <div
                  class="flex flex-row "
                >
                  <h2>
                    <b>
                      1
                    </b>
                  </h2>
                  <i
                    class="icon-calendar text-2xl text-zinc-400"
                  />
                </div>
                <p>
                  route 1
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p>
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-1/12 pl-6 text-right align-middle border-y border-y-light-grey"
        >
          <button
            class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand "
            data-testid="LocatorButton::button"
            type="button"
          >
            <i
              class="icon-bus-alt inline text-center"
            />
          </button>
        </td>
        <td
          class="w-1/12 border-r p-6 text-right align-middle border-y border-y-light-grey"
        >
          <div
            class="group relative flex justify-center"
          >
            <span
              class="absolute z-10 w-max scale-0 rounded-md bg-dark-grey p-2 text-xs 
        text-white transition-all group-hover:scale-100 bottom-11 delay-700"
            >
              Näytä kartalla
            </span>
            <button
              class="text-center h-10 w-10 rounded-full border border-grey bg-white text-tweaked-brand hover:border-2 hover:border-tweaked-brand !bg-background opacity-70 pointer-events-none "
              data-testid="RouteTableRow::showRoute"
              disabled=""
              type="button"
            >
              <svg
                class="text-2xl inline text-center"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
